project1/src/Pages/NewArticle.tsx

```diff
@@ -1,108 +1,116 @@
+> = props => {
+  const [articleTitle, setArticleTitle] = useState("");
+  const [articleDescription, setArticleDescription] = useState("");
+  const [articleBody, setArticleBody] = useState("");
+  const [tagsList, setTagsList] = useState([] as string[]);
+  const slug = props.match.params.slug;
 
-  handleArticleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState({ articleTitle: event.target.value });
+  const handleArticleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setArticleTitle(event.target.value);
   };
 
-  handleArticleDescription = (event: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState({ articleDescription: event.target.value });
+  useEffect(() => {
+    if (!!slug) {
+      AXIOS.noauthGet(`articles/${slug}`).then(res => {
+        setArticleTitle(res.data.article.title);
+        setArticleDescription(res.data.article.description);
+        setArticleBody(res.data.article.body);
+        setTagsList(res.data.article.tagList);
+      });
+    }
+  }, [slug]);
+
+  const handleArticleDescription = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setArticleDescription(event.target.value);
   };
 
-  handleArticleBody = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
-    this.setState({ articleBody: event.target.value });
+  const handleArticleBody = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setArticleBody(event.target.value);
   };
 
-  handleEnterForm = (event: React.KeyboardEvent<HTMLFormElement>) => {
+  const handleEnterForm = (event: React.KeyboardEvent<HTMLFormElement>) => {
     if (event.key === "Enter") event.preventDefault();
   };
 
-  handleAddTag = (tag: string) => {
-    this.setState({
-      tagsList: [...this.state.tagsList, tag]
-    });
+  const handleAddTag = (tag: string) => {
+    setTagsList([...tagsList, tag]);
   };
 
-  handleDeleteTag = (tagID: string) => {
-    const newSelectedItem = this.state.tagsList.filter(id => id !== tagID);
-    this.setState({ tagsList: newSelectedItem });
+  const handleDeleteTag = (tagID: string) => {
+    const newSelectedItem = tagsList.filter(id => id !== tagID);
+    setTagsList(newSelectedItem);
   };
   // handleAddTag:TagsProps["onAddTag"] = (tag)=>{
   //   tag.
   // }
 
-  handleFormSubmition = (event: React.FormEvent<HTMLFormElement>) => {
+  const handleFormSubmition = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     AXIOS.post({
       endpoint: "articles",
       body: {
         article: {
-          title: this.state.articleTitle,
-          description: this.state.articleDescription,
-          body: this.state.articleBody,
-          tagList: this.state.tagsList
+          title: articleTitle,
+          description: articleDescription,
+          body: articleBody,
+          tagList: tagsList
         }
       }
     }).then(data => (window.location.href = `/article/${data.article.slug}`));
```

Comment:
add a ticket to convert all the `window.location.href` routing

Reply:
I already converted them, I will add another PR today